src/streaming/text/DVBFonts.js

```diff
@@ -0,0 +1,258 @@
+ *  without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS AS IS AND ANY
+ *  EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ *  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ *  IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ *  INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ *  NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ *  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ *  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+import Constants from '../constants/Constants';
+import FactoryMaker from '../../core/FactoryMaker';
+import URLUtils from '../utils/URLUtils';
+import EventBus from '../../core/EventBus';
+import Events from '../../core/events/Events';
+import Debug from '../../core/Debug';
+
+function DVBFonts(config) {
+
+    let context = this.context;
+    const eventBus = EventBus(context).getInstance();
+    const urlUtils = URLUtils(context).getInstance();
+    const adapter = config.adapter;
+    const baseURLController = config.baseURLController;
+
+    const FONT_DOWNLOAD_STATUS = {
+        ERROR: 'error',
+        LOADED: 'loaded',
+        UNLOADED: 'unloaded'
+    };
+
+    let instance,
+        logger,
+        dvbFontList;
+
+    function setup() {
+        logger = Debug(context).getInstance().getLogger(instance);
+        resetInitialSettings();
+    }
+
+    /**
+     * Add any dvb fonts from a single track to the dvbFontList
+     * @param {object} track - A text track
+     * @param {string} streamId - Id of current stream
+     * @private
+     */
+    function _addFontFromTrack(track, streamId) {
+        let asBaseUrl;
+        let essentialProperty = false;
+        let dvbFontProps;
+
+        // If there is a baseurl in the manifest resolve against a representation inside the current adaptation set
+        if (baseURLController.resolve()) {
+            const reps = adapter.getVoRepresentations(track);
+            asBaseUrl = baseURLController.resolve(reps[0].path).url
+        }
+
+        const essentialTags = track.essentialPropertiesAsArray.filter(tag => 
+            (tag.schemeIdUri && tag.schemeIdUri === Constants.FONT_DOWNLOAD_DVB_SCHEME)
+        );
+        const supplementalTags = track.supplementalPropertiesAsArray.filter(tag => 
+            (tag.schemeIdUri && tag.schemeIdUri === Constants.FONT_DOWNLOAD_DVB_SCHEME)
+        );
+
+        // When it comes to the property descriptors it's Essential OR Supplementary, with Essential taking preference
+        if (essentialTags.length > 0) {
+            essentialProperty = true;
```

Comment:
Suggest to rename to `isEssential` or `isEssentialProperty`

Reply:
Renamed to `isEssential`